### PR TITLE
Added support for Python 2 ellipsis shortcut for untyped arguments

### PR DESCRIPTION
**Pull request by wittekm**
Implemented #1234 ( support `# type: (...) -> Rtype` ) and happened to throw in a more verbose TypeParseError as well.

Think I should change `TypeParseError().message` to `.reason`? Or strip out the verbose-parse-error entirely?

### CODE REVIEW

**Comment by gvanrossum** [comment]
From a quick skim this looks good. Maybe @jukka or @ddfisher can do a more careful review and answer your questions?

File: mypy/parsetype.py [hunk 1]
```diff
@@ -235,6 +237,15 @@ def parse_signature(tokens: List[Token]) -> Tuple[CallableType, int]:
...
         arg_types.append(arg)
         next = tokens[i].string
+
+        # Check for ellipsis. If it exists, assert it's the only arg_type.
+        # Disallow '(..., int) -> None' for example.
+        if isinstance(arg, EllipsisType):
+            encountered_ellipsis = True
+        if encountered_ellipsis and len(arg_types) != 1:
+            raise TypeParseError(tokens[i], i, "Ellipses cannot accompany other"
+                                               " argument types.")
```
Comment (ddfisher) [inline mypy/parsetype.py]: I would add "... in function type signature." to this error message.

File: mypy/test/data/check-functions.test [hunk 2]
```diff
@@ -1161,6 +1161,45 @@ f(g1)
 f(g2)
 f(g3)
 
+-- (...) -> T
+-- ----------------
+[case testEllipsisWithArbitraryArgsOnBareFunction]
+def f(x, y, z): # type: (...) -> None
```
Comment (ddfisher) [inline mypy/test/data/check-functions.test]: I'd add a couple things to these (non-error) tests:
- a call to the function that shows that you can actually pass it arguments
- arguments with defaults and kwarg-only args

File: mypy/parse.py [hunk 3]
```diff
@@ -1845,7 +1856,8 @@ def parse_error_at(self, tok: Token, skip: bool = True) -> None:
         elif isinstance(tok, Indent) or isinstance(tok, Dedent):
             msg = 'Inconsistent indentation'
         else:
-            msg = 'Parse error before {}'.format(token_repr(tok))
+            reason = " due to: {}".format(reason) if reason else ""
```
Comment (ddfisher) [inline mypy/parse.py]: I'd remove "due to" here.  I think the ":" alone corresponds a bit better to standard error messaging.

Comment (ddfisher) [inline mypy/parse.py]: And as a minor quibble I think I would use a different name here instead of reassigning to `reason`.

**Comment by ddfisher** [comment]
Made a few (reasonably minor) comments.  It's looking good otherwise!

Your changes to `TypeParseError` seem fine to me.

**Comment by wittekm** [comment]
Tweaked per ddfisher.

**Comment by ddfisher** [comment]
LGTM, thanks!

**Pull request merged**